src/lib/mongoBackend/mongoSubCache.cpp

```diff
@@ -839,9 +842,9 @@ int mongoSubCacheItemInsert(const char* tenant, const BSONObj& sub, const char*
   std::vector<BSONElement>  attrVec       = sub.getField(CSUB_ATTRS).Array();
   std::vector<BSONElement>  condVec       = sub.getField(CSUB_CONDITIONS).Array();
 
-  if ((lastNotificationTime == -1) && (sub.hasField(CSUB_LASTNOTIFICATION)))
+  if (lastNotificationTime == -1)
   {
-    lastNotificationTime = sub.getField(CSUB_THROTTLING).Long();
+    lastNotificationTime = getLongField(sub, CSUB_LASTNOTIFICATION);
```

Comment:
Not sure but... maybe this is other case in which "CSUB_LASTNOTIFICATION might be either Int or Long ..." and we need the "dual read" you have implemented in other places to be safer?

(In fact, the "dual read" function seems is used in a lot of places, I'd suggest to move it to a separate and common function)